packages/odyssey-react-mui/src/OdysseyCacheProvider.tsx

```diff
@@ -51,10 +52,16 @@ const OdysseyCacheProvider = ({
       key: uniqueAlphabeticalId,
       nonce: nonce ?? window.cspNonce,
       prepend: true,
-      speedy: false, // <-- Needs to be set to false when shadow-dom is used!! https://github.com/emotion-js/emotion/issues/2053#issuecomment-713429122
+      speedy: shadowDomElement ? false : true, // <-- Needs to be set to false when shadow-dom is used!! https://github.com/emotion-js/emotion/issues/2053#issuecomment-713429122
```

Comment:
If you haven't already, let's post in that issue and see if this has been resolved or not.

And as we already talked about, this unused prop is getting deprecated in favor of `hasShadowDom`.

Reply:
Should I go with `hasShadowDom` or `hasShadowDomElement`?

Reply:
I opted for `hasShadowDom` because it's terser and conveys the same idea more accurately. (For the purposes of caching, we don't care if a shadow DOM element exists, we care if it's using the shadow DOM at all.)

Reply:
I've followed up on the emotion PR here: https://github.com/emotion-js/emotion/issues/2053#issuecomment-2198633837